### ISSUE TITLE
[css-syntax][selectors] Clarify handling of invalid selectors in css-syntax.

### ISSUE THREAD

**Issue by Ms2ger** [w3c/csswg-drafts#85]
CC @tabatkins @SimonSapin 

https://drafts.csswg.org/css-syntax/#css-stylesheets says

> The prelude of the qualified rule is parsed as a [selector list](https://drafts.csswg.org/selectors-4/#selector-list). If this results in an [invalid selector list](https://drafts.csswg.org/selectors-4/#invalid), the entire style rule is invalid.

The prelude of a [qualified rule](https://drafts.csswg.org/css-syntax/#qualified-rule) is a list of component values, and I can't find any prose that defines how to parse that into a selector.

(Pedantic, I know, but this is much less well-defined than the rest of the otherwise well-written spec.)

**Comment by SimonSapin** [comment]
I think the proper way to fix this is to rewrite https://drafts.csswg.org/selectors/#grammar in terms of css-syntax.

Could railroad diagrams be normative?

**Comment by tabatkins** [comment]
Yeah, there's no proper normative description yet in Syntax terms; I have a long-delayed task to rewrite the Selectors grammar accordingly.

Railroad diagrams probably could be normative, if you wanted.  There's no error-correction, so they're fine.  But it's also easy to just write some syntax for it.

**Comment by tabatkins** [comment]
Ah, I did get around to writing the Selectors grammar, so I can rephrase this to explicitly reference the correct productions.

**Comment by tabatkins** [comment]
Fixed.

**Comment by Ms2ger** [comment]
https://github.com/w3c/csswg-drafts/commit/f02bb2e13f30e55539f29bf71715d59599f7d1da
